java/CMakeLists.txt

```diff
@@ -74,6 +74,20 @@ if(BUILD_TESTING)
                                                   GTest::gtest_main)
 endif()
 
+# If the user hasn't set ARROW_JAVA_JNI_ARCH_DIR, derive the normalized
```

Comment:
In general, should the user specify this?
Can we use auto detection as the recommended approach? If the auto detection failed, the user can specify this explicitly as a workaround.

Reply:
I'll change the phrasing of this comment to suggest that the user shouldn't normally need to set this.